### PR TITLE
Clang test

### PR DESCRIPTION
**Pull request by krupitskas**

### CODE REVIEW

File: VorpalEngine/sources/video/Renderer.cpp [hunk 1]
```diff
@@ -18,1062 +18,1133 @@ namespace vp
...
+
+        m_pWindow = glfwCreateWindow(settings.GetApplicationWidth(),
+                                     settings.GetApplicationHeight(),
+                                     settings.GetApplicationName().c_str(),
+                                     nullptr,
+                                     nullptr);
+        glfwSetWindowUserPointer(m_pWindow, this);
+        glfwSetWindowSizeCallback(m_pWindow, onWindowResized);
+        if (!CreateInstance() || !SetupDebugCallback() || !CreateSurface() || !PickPhysicalDevice()
+            || !CreateLogicalDevice()
```
Comment (krupitskas) [inline VorpalEngine/sources/video/Renderer.cpp]: говнясто выглядит это

File: VorpalEngine/sources/video/Renderer.cpp [hunk 2]
```diff
@@ -18,1062 +18,1136 @@ namespace vp
...
-    // Diagnostic info from the Vulkan loader and layers
-    // Usually not helpful in terms of API usage, but may help to debug layer and loader problems
-    if (flags & VK_DEBUG_REPORT_DEBUG_BIT_EXT)
-    {
-        LOG_INFO("VULKAN LAYER DEBUG: %s",  msg);
+        // Error that may result in undefined behaviour
+        if (flags & VK_DEBUG_REPORT_ERROR_BIT_EXT)
+        {
+            LOG_ERROR("VULKAN LAYER ERROR: %s", msg);
+        };
```
Comment (MrDmitry) [inline VorpalEngine/sources/video/Renderer.cpp]: Just noticed that if scope ends with ; :100:

File: VorpalEngine/sources/video/Renderer.cpp [hunk 3]
```diff
@@ -18,1062 +18,1136 @@ namespace vp
...
+        , m_vkInstance(VK_NULL_HANDLE)
+        , m_vkPhysicalDevice(VK_NULL_HANDLE)
+        , m_validationLayers({ "VK_LAYER_LUNARG_standard_validation" })
+        , m_deviceExtensions({ VK_KHR_SWAPCHAIN_EXTENSION_NAME })
     {
-        return false;
+        m_vkSwapChain      = VK_NULL_HANDLE;
+        m_renderPass       = VK_NULL_HANDLE;
+        m_pipelineLayout   = VK_NULL_HANDLE;
+        m_graphicsPipeline = VK_NULL_HANDLE;
```
Comment (MrDmitry) [inline VorpalEngine/sources/video/Renderer.cpp]: Not a huge fan of aligning operations. I prefer only comments, but that doesn't bother me much

File: VorpalEngine/sources/video/Renderer.cpp [hunk 4]
```diff
@@ -18,1062 +18,1136 @@ namespace vp
...
-        details.formats.resize(formatCount);
-        vkGetPhysicalDeviceSurfaceFormatsKHR(device, m_vkWindowSurface, &formatCount, details.formats.data());
+        if (m_pWindow)
+        {
+            glfwSetWindowShouldClose(m_pWindow, GLFW_TRUE);
+            glfwDestroyWindow(m_pWindow);
+            m_pWindow = nullptr;
+        }
+        if (!m_isInitialized)
+            LOG_INFO("Vulkan render shutdown correctly.");
```
Comment (MrDmitry) [inline VorpalEngine/sources/video/Renderer.cpp]: Force braces if possible

File: VorpalEngine/sources/video/Renderer.cpp [hunk 5]
```diff
@@ -18,1062 +18,1136 @@ namespace vp
...
+        else
+        {
+            VkExtent2D actualExtent = { settings.GetApplicationWidth(), settings.GetApplicationHeight() };
 
-    VkInstanceCreateInfo createInfo = {};
-    createInfo.sType = VK_STRUCTURE_TYPE_INSTANCE_CREATE_INFO;
-    createInfo.pApplicationInfo = &appInfo;
-    auto extensions = GetRequiredExtensions();
+            actualExtent.width = std::max(capabilities.minImageExtent.width,
+                                          std::min(capabilities.maxImageExtent.width, actualExtent.width));
```
Comment (MrDmitry) [inline VorpalEngine/sources/video/Renderer.cpp]: Could you reset these split lines to just go to newline with addition indent? Not to the operands level

File: VorpalEngine/sources/video/Renderer.cpp [hunk 6]
```diff
@@ -18,1062 +18,1136 @@ namespace vp
...
+        VkSwapchainCreateInfoKHR createInfo = {};
+        createInfo.sType                    = VK_STRUCTURE_TYPE_SWAPCHAIN_CREATE_INFO_KHR;
+        createInfo.surface                  = m_vkWindowSurface;
+        createInfo.minImageCount            = imageCount;
+        createInfo.imageFormat              = surfaceFormat.format;
+        createInfo.imageColorSpace          = surfaceFormat.colorSpace;
+        createInfo.imageExtent              = extent;
+        createInfo.imageArrayLayers         = 1;
+        createInfo.imageUsage
+            = VK_IMAGE_USAGE_COLOR_ATTACHMENT_BIT; // VK_IMAGE_USAGE_TRANSFER_DST_BIT for post processing.
```
Comment (MrDmitry) [inline VorpalEngine/sources/video/Renderer.cpp]: That looks bad. Comment should've been transferred to the next line, not the assignment

**Review by MrDmitry — CHANGES_REQUESTED** [review]
Could you also try the SimpleStorageTest.cpp, I think we had some inconsistency with it the last time

**Review by MrDmitry — APPROVED** [review]

**Pull request merged**